django-test-suite: iendswith lookup isn't case-insensitive

Test failure:
```
======================================================================
FAIL: test_ticket8597 (queries.tests.ComparisonTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmpfs/src/github/python-spanner-django/django_tests_3/django/tests/queries/tests.py", line 1984, in test_ticket8597
    self.assertQuerysetEqual(
  File "/tmpfs/src/github/python-spanner-django/django_tests_3/django/django/test/testcases.py", line 1056, in assertQuerysetEqual
    return self.assertEqual(list(items), values, msg=msg)
AssertionError: Lists differ: [] != ['<Item: a_b>']

Second list contains 1 additional elements.
First extra element 0:
'<Item: a_b>'

- []
+ ['<Item: a_b>']
```

The generated query is missing `(?i)` in the parameter:

```sql
SELECT ...
FROM queries_item INNER JOIN queries_note ON (queries_item.note_id = queries_note.id) 
WHERE REGEXP_CONTAINS(CAST(queries_item.name AS STRING), %s) 

PARAMS = ('A_b$',);
```

Regression in ff3f43d897c54e3a678811d4f076956975ec721a.

--- a/django_spanner/lookups.py
+++ b/django_spanner/lookups.py
@@ -84,9 +84,9 @@ def startswith_endswith(self, compiler, connection):
             params[0] = str(params[0][1:]) + '$'
         else:
             params[0] = '^' + str(params[0][:-1])
-            # Add the case insensitive flag for istartswith or iendswith.
-            if is_insensitive:
-                params[0] = '(?i)' + params[0]
+        # Add the case insensitive flag for istartswith or iendswith.
+        if is_insensitive:
+            params[0] = '(?i)' + params[0]
         # rhs_sql is REGEXP_CONTAINS(%s, %%s), and lhs_sql is the column name.
         return rhs_sql % lhs_sql, params
     else:

